modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -676,6 +676,10 @@ public OperationsMap postProcessOperationsWithModels(OperationsMap objs, List<Mo
         if (operations != null) {
             final List<CodegenOperation> ops = operations.getOperation();
             for (final CodegenOperation operation : ops) {
+                if (operation.allParams.stream().anyMatch(it -> it.isEnum)) {
+                    objs.getImports().add(importFromMapping("JsonCreator"));
+                    objs.getImports().add(importFromMapping("JsonValue"));
```

Comment:
I doubt that `@JsonCreator` could handle enums as request parameter for cases when `value != enum contant name`(i.e. `AVAILABLE("available")` )

You can check this comment https://github.com/OpenAPITools/openapi-generator/issues/3337#issuecomment-785433635

What I'm thinking is that you have to leverage this PR changes https://github.com/OpenAPITools/openapi-generator/pull/13349

I mean that all these enums should be added to that configuration file